### ISSUE TITLE
package distro subdir naming convention

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#117]
**Reported by efocht on 14 Dec 2004 15:31 UTC**
```
Hi, 
 
the "setup" package API script leaves complete freedom for the 
place where the distribution and arch specific RPMs for a 
package should come from. The result is ever and ever the 
same setup script and varying distro/* names. Now that ia64 
support is really there and x86_64 maybe underway, these 
subdirectory names should even be extended by the 
architecture. 
 
In order to prevent a chaotic development of package 
subdirectory names it would be good to have a reasonable 
convention for these. This would make it easier to add packages 
when the number of supported architectures and distributions 
grows. I propose the usage of the results returned by the 
which_distro_server function in lib/OSCAR/Distro.pm. The 
standard packages/*/distro/ subdirectory name could be 
calculated by something like: 
sub distrodir { 
    my $arch = `uname -i`;  chop $arch; 
    my ($name, $ver) = 
OSCAR::Distro::which_distro_server(); 
    $ver =~ tr/.//; 
    return $name . $ver . "-" . $arch; 
} 
 
This would lead to names like: 
redhat3as-ia64 
redhat80-i386 
fedora2-i386 
etc... 
 
The often duplicated code from the different setup routines 
should then use a generic-setup script located in scripts/ . The 
attached file could fulfill this functionality. The setup scripts 
would then be simplified to: 
pvm/scripts/setup: 
#!/bin/sh 
$OSCAR_HOME/scripts/generic-setup -e pvm 
 
switcher/scripts/setup: 
#!/bin/sh 
$OSCAR_HOME/scripts/generic-setup modules-oscar 
 
ganglia/scripts/setup: (doesn't exist yet, but actually should...) 
#!/bin/sh 
$OSCAR_HOME/scripts/generic-setup -e rrdtool 
ganglia-oscar-monitor-core-gmond 
ganglia-monitor-core-gmetad 
  
etc... 
 
The RPMs which are _really_ common (usually: noarch) could 
continue to live in packages/*/RPMS. 
 
Regards, 
Erich 
 
```

**Comment by dikim33** [comment]
**Comment by bli on 1 Sep 2005 07:22 UTC**
```

Assigning to Erich since he has been taking the lead on this :-)
```

**Comment by dikim33** [comment]
**Comment by efocht on 18 Jul 2006 22:18 UTC**
This has been implemented and is used in OSCAR 5.0.
Closing.
